source/python/run_scraper.py

```diff
@@ -1,42 +1,44 @@
+import click
 import logging
-import sys
 from datetime import timedelta
 
-from sqlalchemy import create_engine, func
+from sqlalchemy import func
 
-from idetect.model import db_url, Base, Session, Status, Analysis
+from idetect.configs import Command
+from idetect.model import Status, Analysis
 from idetect.scraper import scrape
-from idetect.worker import Worker
 
 MAX_RETRIEVAL_ATTEMPTS = 3
 HOURS_BETWEEN_ATTEMPTS = 12
 
 
+logger = logging.getLogger(__name__)
+
+
 # Filter function for identifying analyses to scrape
 def scraping_filter(query):
     # Choose either New analyses OR
     # Analyses where Scraping Failed &
     # less than 3 scraping attempts &
     # last scraping attempt greater than 12 hours ago
-    return query.filter((Analysis.status == Status.NEW) |
-                        ((Analysis.status == Status.SCRAPING_FAILED) &
-                         (Analysis.retrieval_attempts < MAX_RETRIEVAL_ATTEMPTS) &
-                         (func.now() > Analysis.retrieval_date + timedelta(hours=HOURS_BETWEEN_ATTEMPTS))))
-
-
-if __name__ == "__main__":
-    logger = logging.getLogger(__name__)
-    logger.setLevel(logging.INFO)
-    handler = logging.StreamHandler(sys.stderr)
-    handler.setFormatter(logging.Formatter("%(asctime)s %(message)s"))
-    logger.root.addHandler(handler)
-
-    engine = create_engine(db_url())
-    Session.configure(bind=engine)
-    Base.metadata.create_all(engine)
-
-    worker = Worker(scraping_filter, Status.SCRAPING, Status.SCRAPED, Status.SCRAPING_FAILED,
-                    scrape, engine)
-    logger.info("Starting worker...")
-    worker.work_indefinitely()
-    logger.info("Worker stopped.")
+    return query.filter(
+        (Analysis.status == Status.NEW) |
+        (
+            (Analysis.status == Status.SCRAPING_FAILED) &
+            (Analysis.retrieval_attempts < MAX_RETRIEVAL_ATTEMPTS) &
+            (func.now() > Analysis.retrieval_date + timedelta(hours=HOURS_BETWEEN_ATTEMPTS))
+        )
+    )
```

Comment:
Just curious on how this works? Is this bitwise operation?
`filter` generally takes function as argument to filter each item.

Reply:
Yeah, basically this condition is changed to SQL. Something like this...

```SQL
SELECT * ...
WHERE STATUS = 'NEW' OR (
  STATUS = 'SCRAPING_FAILED' AND RETRIVAL_ATTEMPTS SOME-INTEGER-NUMBER AND ....
)
```

Reply:
It's weird how they do that.
Is'nt return type for `Analysis.status == Status.NEW` a boolean?
Does python support operator overloading?

Reply:
Not sure will need to look in-depth https://github.com/sqlalchemy/sqlalchemy
In Django ORM we pass as named attributes like (.filter(status=Status.NEW, ... ).

Reply:
Looks like they do overload `==` operator

Reply:
Oh... so sqlalchemy is overloading the operators?... Nice...